### ISSUE TITLE
Auto-fix wrongly suggesting scope issues during struct manipulations 

### ISSUE THREAD

**Issue by jerryajay** [rust-lang/rust#71027]
Hi, there's a auto-fix bug. I'm using VSCode. The code explains itself:

I tried this code:

```rust
use std::fmt;
use std::fmt::Error;
use std::fmt::Formatter;

struct Structure(i32);

#[derive(Debug)]
struct Point {
    x : i32,
    y : i32,
}

impl fmt::Display for Structure {
    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
        write!(f, "The value inside the structure is: {}", self.0)
    }
}


fn main() {
    let s = Structure(12);
    println!("{}", s);
    let point = {x = 12; y = 45};
    println!("{}")
}
```

I expected to see the auto-fix suggest ways to print Point correctly.

Instead, auto-fix in VSCode suggests changing x and y to s in line 23.
However, the error message from rust displays fine.

### Meta

`rustc --version --verbose`:
```
<version>
```

<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by jonas-schievink** [comment]
It's not really clear what this issue is about. Are you using RLS? I'm assuming it's displaying the suggestions in this error message?

```
error[E0425]: cannot find value `x` in this scope
  --> src/main.rs:23:18
   |
23 |     let point = {x = 12; y = 45};
   |                  ^ help: a local variable with a similar name exists: `s`
```

If so, that is definitely a correct error message and `help`, even if the result is not what you wanted (`help` is only a suggestion, not a solution, after all).

Maybe we can add extra diagnostics for syntax that looks like a C struct initializer, but I don't think too many people hit this.

**Comment by jerryajay** [comment]
Yes, RLS is being used. Rust is powerful for a blatant help message like this. I'll submit a pull request for the extra diagnostics - time permitting.

**Comment by fmease** [comment]
> Maybe we can add extra diagnostics for syntax that looks like a C struct initializer,

`{ x = …; y = … }` looks more like an OCaml record expressions than a C struct initializer `{ .x = …, .y = … }`.
